core/src/Revolution/Processors/Workspace/Lexicon/Create.php

```diff
@@ -43,12 +44,14 @@ public function getLanguageTopics()
      */
     public function process()
     {
-        if ($this->alreadyExists()) {
+        $data = array_map('trim', $this->getProperties());
```

Comment:
This may cause issues as it trims values as well. Some lexicons end with a space by design, because they're suffixed (in code) by a value. Some quick examples I found:

```
$_lang['file_err_open'] = 'Cannot open file: ';
$_lang['package_update_to_version'] = 'Please select the version to update this package to: ';
$_lang['error_sending_email_to'] = 'Error while sending mail to ';
```

I'd suggest removing the array_map here and specifically trimming the namespace, topic, and language if needed in alreadyExists/fromArray.

Reply:
OK, yes, I didn't think of that! I'll update that later today...

Reply:
@Mark-H it makes sense to change lexicon items in future:
```php
$_lang['file_err_open'] = 'Cannot open file: ';
```
to
```php
$_lang['file_err_open'] = 'Cannot open file: [[+file]]';
```

Respectively:
```php
$this->modx->lexicon('file_err_open') . $this->getProperty('file');
```
to

```php
$this->modx->lexicon('file_err_open', [ 'file' => $this->getProperty('file') ]);
```

This will allow not to worry about such dangling lexicon items.

Reply:
I don't disagree, but it's probably better to leave that for another day ;)